strutil/quantity/quantity.go

```diff
@@ -200,3 +201,248 @@ func FormatDuration(dt float64) string {
+	}
+
+	// Special case: render as "ages!"
+	if dt >= float64(math.MaxUint64) {
+		return "ages!"
+	}
+
+	// Fractional seconds to nanoseconds
+	delta := Duration(dt * float64(Second))
+
+	// If we only render a subset of place values, we need to apply
+	// the render mode to the last place value. We do this indirectly
+	// by moving up the 'min' accuracy.
+	if count == ShowCompact {
+		for i, key := range keys {
+			// Compact mode only renders two places up until seconds
+			// The indexing is safe as places beyond seconds exist
+			if delta >= key && key >= Minute {
+				index := i + 1
+				if keys[index] > min {
+					min = keys[index]
+				}
+				break
+			}
+
+			// Only one place if we are s, ms, us or ns
+			if delta >= key && key < Minute {
+				if key > min {
+					min = key
+				}
+				break
+			}
+		}
+	}
+
+	// Apply the rendermode to the least significant place value
+	mod := delta % min
+	switch {
+	case mode == TimeLeft && mod > 0:
+		// Ceiling
+		delta = delta + min - mod
+	case mode == TimePassed:
+		// Floor
+		delta = delta - mod
+	case mode == TimeRounded && mod >= (min/2):
+		// Rounded Up
+		delta = delta + min - mod
+	case mode == TimeRounded && mod < (min/2):
+		// Rounded Down
+		delta = delta - mod
+	}
+
+	// Special case: less than minimum accuracy
+	if delta < min {
+		return "0" + units[min]
+	}
+
+	// In ShowCompact mode only 2-digit days are possible
+	if count == ShowCompact && delta >= (100*Day) {
+		return "ages!"
+	}
+
+	// Iterate through units of time in order from
+	// highest to lowest (years, days, ...)
+	for _, key := range keys {
+		done := false
+
+		// Nothing left to render, do not bother iterating any further
+		if delta <= 0 {
+			break
+		}
+
+		// No place value left greater than required
+		// accuracy to render.
+		if delta < min && len(render) != 0 {
+			break
+		}
+
+		// Remainder is less than current place value,
+		// skip to next.
+		if delta < key {
+			continue
+		}
+
+		// If the duration exceeds the maximum place value
+		// (unit if time) we will not render it, just return
+		// "ages!".
+		if key > max {
+			return "ages!"
+		}
+
+		unit := delta / key
+		remainder := delta % key
+
+		// If the accuracy is less than 1s, we support
+		// rendering a fractional second part.
+		if delta < Minute && remainder > 0 && count == ShowVerbose {
+			digits := 0
+
+			// Number of digits required to render
+			// nano second fractunal accuracy.
+			for cc := key; cc > 1; cc /= 10 {
+				digits++
+			}
+
+			// Number of digits to render specified
+			// fractunal accuracy.
+			for cc := min; cc > 1; cc /= 10 {
+				digits--
+			}
+
+			frac := float64(delta) / float64(key)
+
+			// Should we generate a fractional second, else
+			// use the generic rendering function.
+			if digits > 0 {
+				if len(render) > 0 && space == SpaceOn {
+					render += " "
+				}
+				render += strconv.FormatFloat(frac, 'f', digits, 64)
+				render += units[key]
+
+				delta = 0
+				done = true
```

Comment:
it seems to me that this `done` variable is not needed, and that we could just replace it with `delta == 0` in the `if` conditions

Reply:
Yes! Fixed.